files/galaxy/dynamic_job_rules/production/total_perspective_vortex/default_tool.yml.j2

```diff
@@ -65,15 +66,33 @@ tools:
 
           def destination_usage_proportion(destination):
             if not destination.context.get('destination_total_mem') or not destination.context.get('destination_total_cores'):
-                    raise Exception(f"++ ---tpv rank debug: At least one of destination_total_mem, destination_total_cores is unavailable")
+              raise Exception(f"++ ---tpv rank debug: At least one of destination_total_mem, destination_total_cores is unavailable")
             destination_total_cores = destination.context.get('destination_total_cores')
             return_value = sum([db_queue_info.get(destination.id, {}).get(state, {}).get('sum_cores', 0) for state in ['queued', 'running']])/destination_total_cores
             log.info(f"++ ---tpv rank debug: returning usage proportion value for destination {destination.id}: {str(return_value)}")
             return return_value
 
+          def destination_availability_score(destination):
+              if not destination.context.get('destination_total_mem') or not destination.context.get('destination_total_cores'):
+                raise Exception(f"++ ---tpv rank debug: At least one of destination_total_mem, destination_total_cores is unavailable")
+              destination_cores_committed = sum([db_queue_info.get(destination.id, {}).get(state, {}).get('sum_cores', 0) for state in ['queued', 'running']])
+              destination_mem_committed = sum([db_queue_info.get(destination.id, {}).get(state, {}).get('sum_mem', 0) for state in ['queued', 'running']])
+              available_cores = destination.context['destination_total_cores'] - destination_cores_committed
+              available_mem = float(destination.context['destination_total_mem']) - (float(destination_mem_committed)/1024)
```

Comment:
So, `destination_mem_committed` and `destination.context['destination_total_mem']` are using a different base (i.e. KB vs GB)?

Reply:
because the query is taking the value from slurm native specification, which is in MB